src/main/scala/org/broadinstitute/dsde/workbench/sam/directory/PostgresDirectoryDAO.scala

```diff
@@ -154,7 +154,67 @@ class PostgresDirectoryDAO(protected val dbRef: DbReference,
     */
   override def removeGroupMember(groupId: WorkbenchGroupIdentity, removeMember: WorkbenchSubject): IO[Boolean] = ???
 
-  override def isGroupMember(groupId: WorkbenchGroupIdentity, member: WorkbenchSubject): IO[Boolean] = ???
+  override def isGroupMember(groupId: WorkbenchGroupIdentity, member: WorkbenchSubject): IO[Boolean] = {
+    val gm = GroupMemberTable.syntax("gm")
+
+    val memberClause: SQLSyntax = member match {
+      case WorkbenchGroupName(groupName) =>
+        val g = GroupTable.syntax("g")
+        sqls"sg.member_group_id = (select ${g.id} from ${GroupTable.as(g)} where ${g.name} = $groupName)"
+
+      case FullyQualifiedPolicyId(FullyQualifiedResourceId(resourceTypeName, resourceId), accessPolicyName) =>
+        val rt = ResourceTypeTable.syntax("rt")
+        val r = ResourceTable.syntax("r")
+        val p = PolicyTable.syntax("p")
+        sqls"""sg.member_group_id = (select ${p.groupId}
+              from ${ResourceTypeTable.as(rt)}
+              join ${ResourceTable.as(r)} on ${rt.id} = ${r.resourceTypeId}
+              join ${PolicyTable.as(p)} on ${r.id} = ${p.resourceId}
+              where ${rt.resourceTypeName} = $resourceTypeName)
+              and ${r.name} = $resourceId
+              and ${p.name} = $accessPolicyName)"""
+
+      case WorkbenchUserId(userId) => sqls"sg.member_user_id = $userId"
+
+      case _ => throw new WorkbenchException(s"illegal member $member")
+    }
+
+    val topGroupQuery = groupId match {
+      case WorkbenchGroupName(groupName) =>
+        val g = GroupTable.syntax("g")
+        sqls"select ${g.id}, ${gm.memberGroupId}, ${gm.memberUserId} from ${GroupTable.as(g)} join ${GroupMemberTable.as(gm)} on ${g.id} = ${gm.groupId} WHERE ${g.name} = $groupName"
+
+      case FullyQualifiedPolicyId(FullyQualifiedResourceId(resourceTypeName, resourceId), accessPolicyName) =>
+        val rt = ResourceTypeTable.syntax("rt")
+        val r = ResourceTable.syntax("r")
+        val p = PolicyTable.syntax("p")
+        sqls"""select ${p.groupId}, ${gm.memberGroupId}, ${gm.memberUserId}
+              from ${ResourceTypeTable.as(rt)}
+              join ${ResourceTable.as(r)} on ${rt.id} = ${r.resourceTypeId}
+              join ${PolicyTable.as(p)} on ${r.id} = ${p.resourceId}
+              join ${GroupMemberTable.as(gm)} on ${p.groupId} = ${gm.groupId}
+              where ${rt.resourceTypeName} = $resourceTypeName)
+              and ${r.name} = $resourceId
+              and ${p.name} = $accessPolicyName"""
+
+    }
```

Comment:
Similar to:

https://github.com/broadinstitute/sam/blob/4e05af7571fd061b9d93dcb2f90131ecb56f5fbf/src/main/scala/org/broadinstitute/dsde/workbench/sam/directory/PostgresDirectoryDAO.scala#L240

Could we write a method like, `def groupMembersQuery(groupId: WorkbenchGroupIdentity): SQLSyntax = ???`